src/sign/decentralized_party/signature_threshold_decryption_round.rs

```diff
@@ -156,22 +156,33 @@ where
         Ok((self.nonce_x_coordinate, signature_s))
     }
 
-    /// The lightweight $$ O(1) $$ threshold decryption logic, which simply verifies the output of
-    /// the decryption sent by the designated decrypting party. Blames it in case of an invalid
-    /// signature, and accepts otherwise.
+    /// A wrapper function for [`Self::verify_decrypted_signature_wrapper`] that uses self's
+    /// attributes to verify the decrypted signature.
     pub fn verify_decrypted_signature(
         self,
         signature_s: GroupElement::Scalar,
-        designated_decrypting_party_id: PartyID,
     ) -> crate::Result<(GroupElement::Scalar, GroupElement::Scalar)> {
-        verify_signature(
+        Self::verify_decrypted_signature_wrapper(
             self.nonce_x_coordinate,
             signature_s,
             self.message,
             self.public_key,
         )
-        .map_err(|_| Error::MaliciousDesignatedDecryptingParty(designated_decrypting_party_id))?;
+    }
 
-        Ok((self.nonce_x_coordinate, signature_s))
+    /// The lightweight $O(1)$ threshold decryption logic, which simply verifies the output of
+    /// the decryption sent by the designated decrypting party.
+    /// Returns a [`Error::MaliciousDesignatedDecryptingParty`] for an invalid signature,
+    /// and accepts otherwise.
+    pub fn verify_decrypted_signature_wrapper(
```

Comment:
```suggestion
    pub fn verify_decrypted_signature(
```
If anything, this is the internal function so the other one should be called `wrapper`